### ISSUE TITLE
Publish with NO_ACK missing ttl field?

### ISSUE THREAD

**Issue by kennethlimcp** [spark/firmware#1201]
A user tried doing `Particle.publish("t", temperature, ttl, PRIVATE, NO_ACK);`: http://community.particle.io/t/problems-compiling-particle-publish-with-no-ack-on-electron-solved/27924

as per documented here: https://docs.particle.io/reference/firmware/electron/#particle-publish- 

but it seems like it's not working.

Test case on v0.6.0:

```
void setup()
{
  Particle.publish("t", "10", 10, PRIVATE, NO_ACK);
}
```

---

Completeness:
- [x] Minimum test case added
- [x] Device, system and user firmware versions stated

**Comment by m-mcgowan** [comment]
Thanks for the report. 

> but it seems like it's not working.

We need a bit more to do on ;-) Please explain the observed behaviour vs expected behaviour. Thanks!

**Comment by kennethlimcp** [comment]
opps..

```
test.cpp:12:50: error: no matching function for call to 'CloudClass::publish(const char [2], const char [3], int, const PublishFlag&, const PublishFlag&)'
 
                                                  ^

test.cpp:12:50: note: candidates are:
In file included from ../wiring/inc/spark_wiring.h:48:0,
                 from ./inc/application.h:36,
                 from test.cpp:2:
../wiring/inc/spark_wiring_cloud.h:222:17: note: bool CloudClass::publish(const char*, PublishFlag)
     inline bool publish(const char *eventName, PublishFlag eventType=PUBLIC)
                 ^
../wiring/inc/spark_wiring_cloud.h:222:17: note:   candidate expects 2 arguments, 5 provided
../wiring/inc/spark_wiring_cloud.h:227:17: note: bool CloudClass::publish(const char*, const char*, PublishFlag)
     inline bool publish(const char *eventName, const char *eventData, PublishFlag eventType=PUBLIC)
                 ^
../wiring/inc/spark_wiring_cloud.h:227:17: note:   candidate expects 3 arguments, 5 provided
../wiring/inc/spark_wiring_cloud.h:232:17: note: bool CloudClass::publish(const char*, const char*, PublishFlag, PublishFlag)
     inline bool publish(const char *eventName, const char *eventData, PublishFlag f1, PublishFlag f2)
                 ^
../wiring/inc/spark_wiring_cloud.h:232:17: note:   candidate expects 4 arguments, 5 provided
../wiring/inc/spark_wiring_cloud.h:238:17: note: bool CloudClass::publish(const char*, const char*, int, PublishFlag)
     inline bool publish(const char *eventName, const char *eventData, int ttl, PublishFlag eventType=PUBLIC)
                 ^
../wiring/inc/spark_wiring_cloud.h:238:17: note:   candidate expects 4 arguments, 5 provided
make[1]: *** [../build/target/user/platform-6test.o] Error 1
make: *** [user] Error 2

```

Excepted: compiles fine
Actual: compilation failed

**Comment by m-mcgowan** [comment]
The first two arguments are strings - the event name and the event data, so the first `10` should be `"10"`

**Comment by kennethlimcp** [comment]
copy-pasta error. Updated the test case (that still doesn't compile)

**Comment by m-mcgowan** [comment]
Good catch Kenneth...looks like the 5 arg version is missing. This is a quick fix, so we'll get it out in the next release.  Although please keep in mind that I don't believe TTL is actually honoured at present, so the workaround is to simply remove it.

**Comment by kennethlimcp** [comment]
Definitely, but the docs said otherwise. 🗡

**Comment by smpickett** [comment]
There still appears to be a problem with the documentation that needs to be updated. Specifically:

int temperature = sensor.readTemperature();  // by way of example, not part of the API
Particle.publish("t", temperature, NO_ACK);
Particle.publish("t", temperature, PRIVATE, NO_ACK);
Particle.publish("t", temperature, ttl, PRIVATE, NO_ACK);

should be changed so that the 'temperature' argument is a string. No?

**Comment by technobly** [comment]
Yes!  Thanks @smpickett, will do it now.

**Comment by technobly** [comment]
And it's live: https://docs.particle.io/reference/firmware/electron/#particle-publish- and https://prerelease-docs.particle.io/reference/firmware/electron/#particle-publish-

**Comment by JohnKozell** [comment]
I'm compiling to electron and 0.6.2, yet I'm still getting the following error:
`no matching function for call to 'CloudClass::publish(const char [8], char [255], int, const PublishFlag&, const PublishFlag&)`

Am I missing something?

My call is:
`  Particle.publish("reading", payload, 60, PRIVATE, NO_ACK);`
Where payload is defined as:
`char payload[255];`

**Comment by sergeuz** [comment]
@JohnKozell The fix is available starting with the 0.7.0-rc.1 release
